bitcoin/src/taproot.rs

```diff
@@ -1337,7 +1645,7 @@ mod test {
         let builder = builder.finalize(&secp, internal_key).unwrap_err();
         let builder = builder.add_leaf(3, e.clone()).unwrap();
 
-        let tree_info = builder.finalize(&secp, internal_key).unwrap();
+        let tree_info = builder.clone().finalize(&secp, internal_key).unwrap();
```

Comment:
No need to clone here.

Reply:
The test below uses builder so I needed to conditionally clone under #cfg(serde). But I have rearranged the code so that it's much cleaner now.